app.json

```diff
@@ -0,0 +1,58 @@
+{
+  "name": "Portier Broker",
+  "description": "Portier Broker reference implementation, written in Rust. Learn more at https://portier.github.io/.",
+  "keywords": [
+    "Portier",
+    "Passwordless",
+    "Authentication"
+  ],
+  "website": "https://portier.github.io/",
+  "repository": "https://github.com/portier/portier-broker",
+  "env": {
+    "BROKER_PUBLIC_URL": {
+      "description": "The broker server's public-facing URL. This will be the Issuer for JSON Web Tokens and Relying Parties will use the same value for their Broker URL. To ensure consistency, trailing slashes should be avoided. It is suggested that you choose an app name (up above) and then use it to fill out this config var. Alternatively, if you enable the Dyno Metadata labs addon using the Heroku CLI then you can leave this value blank and the public-facing URL will be automatically set, thus allowing you to let Heroku choose the app's name.",
+      "value": "https://[app-name].herokuapp.com"
+    },
+    "BROKER_KEYTEXT": {
+      "description": "The text of an encryption key in PEM format. You can create an encryption key by running `openssl genrsa 4096 > private.pem` and then provide the key to this broker by copying the file's contents here.",
+      "value": ""
+    },
+    "BROKER_FROM_ADDRESS": {
+      "description": "The 'From' address used by Portier to send emails. Unlike the BROKER_PUBLIC_URL, this config value will not auto-populate even if the Dyno Metadata labs addon is enabled. It will be easiest if you choose an app name (up above) and then use it here instead of letting Heroku pick a name for the app.",
+      "value": "portier@[app-name].herokuapp.com"
+    },
+    "BROKER_FROM_NAME": {
+      "description": "The 'From' name on the emails sent by Portier.",
+      "value": "Portier"
+    },
+    "BROKER_ALLOWED_ORIGINS": {
+      "description": "The origins allowed to use this broker for authentication. You can specify a single origin or list multiple origins and separate them with commas.",
+      "value": "https://example.com,https://example.org:8080"
+    },
+    "BROKER_GMAIL_CLIENT": {
+      "description": "Your Google OAuth API Client ID (enables in-browser Google Authentication for Gmail users, if desired)",
+      "value": "",
+      "required": false
+    },
+    "BROKER_GMAIL_SECRET": {
+      "description": "Your Google OAuth API Secret Key. Only required if a corresponding Client ID was provided, above.",
+      "value": "",
+      "required": false
+    }
+  },
+  "formation": {
+    "web": {
+      "quantity": 1,
+      "size": "free"
+    }
+  },
+  "addons": [
+    "heroku-redis:hobby-dev",
+    "sendgrid:starter"
```

Comment:
I think this is worth considering. I've experimented with both Mailgun and SendGrid but have not yet tried Postmark. Emails from SendGrid have usually been arriving in my Gmail inbox within one to two minutes of clicking the login button. While that's not the speed desired for production deployments it feels sufficient for someone doing a test drive or running the broker for their hobby projects.

When I was first putting the app.json together I used Mailgun. Unfortunately the onboarding process for new Mailgun addon instances required additional verification through the Mailgun website before being able to send emails. It seemed overly burdensome for new users seeking an "it just works" workflow to have to visit their Heroku app's dashboard, find the Mailgun addon link, and follow the steps on the Mailgun site before being able to test their newly deployed broker.

According to the [Heroku docs for Postmark](https://devcenter.heroku.com/articles/postmark#creating-sender-signatures), Postmark also requires verification in advance of sending emails. Since I haven't tried it I can't confirm how it works.

Sendgrid has not required similar verification and enables a much smoother deployment. A user can click the 'Deploy to Heroku' button, fill in their values, wait for the dyno to build, point their relying party at the new broker instance, and see immediate success when testing a login.

I think SendGrid is our best option if the 'Deploy to Heroku' button is targeted at hobbyist users and developers wanting to test Portier. Perhaps migrating to other SMTP providers could be described in the post-install documentation for users wanting to create production-ready deployments on Heroku?